packages/client/src/worker/protocol.ts

```
@@ -221,13 +221,25 @@ export interface ReceivedShareDescriptor {
   resolution: ReceivedShareResolution | null;
 }
 
+/**
+ * Where a bin row's origin folder stands in the vault (mirrors the facade
+ * `BinOrigin`). `'gone'` is the state a default restore refuses on, so a host
+ * names it rather than showing a folder that is not there.
+ */
+export type BinOriginDescriptor =
+  | { kind: 'root' }
+  | { kind: 'folder'; name: string }
+  | { kind: 'gone' };
```

ESLint

[error] Replace `|·{·kind:·'root'·}⏎··|·{·kind:·'folder';·name:·string·}⏎·` with `{·kind:·'root'·}·|·{·kind:·'folder';·name:·string·}`

(prettier/prettier)